api/src/main/java/bio/terra/tanagra/service/search/SqlVisitor.java

```diff
@@ -130,12 +133,54 @@ public String visitBinaryComparision(BinaryFunction binaryFunction) {
           return String.format("%s < %s", leftSql, rightSql);
         case EQUALS:
           return String.format("%s = %s", leftSql, rightSql);
+        case DESCENDANT_OF:
+          return resolveDescendantOf(binaryFunction);
         default:
           throw new UnsupportedOperationException(
               String.format("Unsupported BinaryFunction.Operator %s", binaryFunction.operator()));
       }
     }
 
+    /**
+     * Returns an SQL string for a {@link BinaryFunction.Operator#DESCENDANT_OF} filter function.
+     */
+    private String resolveDescendantOf(BinaryFunction binaryFunction) {
+      Preconditions.checkArgument(
+          binaryFunction.operator().equals(BinaryFunction.Operator.DESCENDANT_OF));
+      if (!(binaryFunction.left() instanceof Expression.AttributeExpression)) {
+        throw new BadRequestException("DESCENDANT_OF only supported for attribute left operand.");
+      }
+      Expression.AttributeExpression attributeExpression =
+          (Expression.AttributeExpression) binaryFunction.left();
+      Hierarchy hierarchy =
+          searchContext
+              .underlay()
+              .hierarchies()
+              .get(attributeExpression.attributeVariable().attribute());
+      if (hierarchy == null) {
+        throw new BadRequestException(
```

Comment:
To check my understanding: using DECENDANT_OF with a leaf concept is illegal - right?

Reply:
Ergh, that should probably be fine with this operator does - include the concept and its descendants so it's legal on leaves.
We could do this in the index generation (i.e. everything has the identity relationship), or in our SQL expansion 1(attribute = 1234 OR attribute in (select descendants of 1234))`.
In OMOP's concept_ancestor, each concept is recorded as the ancestor of itself, so that's some kind of precedent. https://www.ohdsi.org/web/wiki/doku.php?id=documentation:cdm:concept_ancestor
Maybe also a better name for the operator? `INCLUDE_DESCENDANTS`? `IS_OR_DESCENDANT_OF`? Thoughts?

Reply:
I recall that the AoU cohort builder actually supports both (select a tree WITH the parent node, or without), but I can't remember if:

1. this is built into the index (i.e. criteria tables), or
2. this is something the user can decide at query time

 @freemabd could you confirm which one it is?

If it's scenario (1), your current operator name makes sense, though arguably this is too opaque. If it's (2), then perhaps we should have two operators OR we just go with what you have now (exclude parent), and leave it to the client to compose the predicate themselves.

Reply:
> I recall that the AoU cohort builder actually supports both (select a tree WITH the parent node, or without), but I can't remember if:
> 
> 1. this is built into the index (i.e. criteria tables), or
> 2. this is something the user can decide at query time

This is built into the criteria table.

Reply:
Per offline discussion with CH - for now we'll make the inclusion of the parent explicit in the name of the operator. Later we can evaluate if we want to make inclusion determined by the index, or support different operators.
@calbach PTAL